easypy/semver.py

```diff
@@ -0,0 +1,187 @@
+from collections import namedtuple
+import re
+
+
+class SemVerParseException(ValueError):
+    pass
+
+
+class SemVer(namedtuple("SemVer", "major minor patch build tag")):
+    """ Semantic Version object
+
+    From https://semver.org:
+    Given a version number MAJOR.MINOR.PATCH, increment the:
+
+    MAJOR version when you make incompatible API changes,
+    MINOR version when you add functionality in a backwards-compatible manner, and
+    PATCH version when you make backwards-compatible bug fixes.
+    Additional labels for pre-release and build metadata are available as extensions to the MAJOR.MINOR.PATCH format.
+
+    We use a fourth part, let's call it BUILD and define it is incremented sporadically.
+    """
+
+    @classmethod
+    def loads(cls, string, *, separator='.', tag_separator='-', raise_on_failure=True):
+        """
+        Load a string into a SemVer object.
+
+        :param string: The string representing the semantic version. Must adhere to semver format.
+        :type string: str
+        :param separator: Use a different version part separator.
+        :type separator: str, optional
+        :param tag_separator: Use a different tag separator.
+        :type tag_separator: str, optional
+        :param raise_on_failure: Whether to raise on failure or just return None
+        :type raise_on_failure: bool, optional
+        """
+
+        string, _, tag = string.partition(tag_separator)
+        parts = string.split(separator)
+        try:
+            return cls(*parts, tag=tag)
+        except ValueError as e:
+            if raise_on_failure:
+                raise SemVerParseException("Error parsing %s: %s" % (string, str(e))) from None
+            else:
+                return None
+
+    @classmethod
+    def loads_fuzzy(cls, string):
+        """
+        Load a string into a SemVer without enforcing semver format.
+
+        :param string: The string representing the semantic version. Must adhere to semver format.
+        :type string: str
+        """
+
+        regex = re.compile(r"((?:\d+[.-])+)(.*)")
+        string, tag = regex.fullmatch(string).groups()
+        parts = re.split("[-.]", string)
+        return cls(*filter(None, parts), tag=tag)
+
+    def __new__(cls, major=0, minor=0, patch=None, build=None, *, tag=None):
+        return super().__new__(
+            cls,
+            major=int(major),
+            minor=int(minor),
+            patch=int(patch) if patch is not None else None,
+            build=int(build) if build is not None else None,
+            tag="" if tag is None else str(tag),
+        )
+
+    @property
+    def __dict__(self):
+        return super().__dict__
```

Comment:
Why is this necessary?

Reply:
seems like it doesn't work anyway. I'll remove this